src/services/formatting/formatting.ts

```diff
@@ -493,9 +480,9 @@ namespace ts.formatting {
                     }
                 },
                 getIndentation: () => indentation,
-                getDelta: () => delta,
+                getDelta: (child) => (!delta || SmartIndenter.shouldInheritParentIndentation(node, child)) ? 0 : delta,

```

Comment:
The delta value increases indentation for child node, so it should be zero when the child node should just copy indentation from parent.